src/hivpy/hiv_testing.py

```diff
@@ -99,46 +102,41 @@ def test_mark_hiv_symptomatic(self, pop):
 
             if len(not_diag_tested_pop) > 0:
                 # mark people for testing
-                marked = pop.transform_group([pop.get_correct_column(col.ADC, dt=1),
-                                              pop.get_correct_column(col.TB, dt=1),
-                                              pop.get_correct_column(col.TB, dt=2),
-                                              pop.get_correct_column(col.NON_TB_WHO3, dt=1)],
+                marked = pop.transform_group([col.ADC, col.TB_INFECTION_DATE, col.NON_TB_WHO3],
                                              self.calc_symptomatic_testing_outcomes,
                                              sub_pop=not_diag_tested_pop)
                 # set outcomes
                 pop.set_present_variable(col.TEST_MARK, marked, not_diag_tested_pop)
```

Comment:
Not sure we can actually use `col.TB_INFECTION_DATE` in `transform_group`; the date is unhashable and it looks like the DataFrame's `groupby` won't work with it.